src/main/java/pl/edu/mimuw/logic/Formula.java

```diff
@@ -1,8 +1,10 @@
 package pl.edu.mimuw.logic;
 
+import java.util.ArrayList;
+
 public interface Formula {
 
   boolean eval(Valuation valuation);
 
-  // TODO: extend interface with needed methods
+  ArrayList<FormulaVariable> variables();
```

Comment:
Stosujmy interfejsy w typach wartości zwracanych przez metody, wiec jak już to `List<FormulaVariable>` ale lepiej to nawet `Set<FormulaVariable>` by wyrazić to, że nie bedzie tam powtorzen